presto-native-execution/presto_cpp/main/PrestoServer.h

```diff
@@ -83,9 +83,16 @@ class PrestoServer {
     return nodeState_;
   }
 
-  void setNodeState(NodeState nodeState) {
-    nodeState_ = nodeState;
-  }
+  /// Set worker into the SHUTTING_DOWN state even if we aren't shutting down.
+  /// This will prevent coordinator from sending new tasks to this worker.
+  void detachWorker();
+
+  /// Set worker into the ACTIVE state if we aren't shutting down.
+  /// This will enable coordinator to send new tasks to this worker.
+  void maybeAttachWorker();
```

Comment:
What does 'maybe' mean here? Would you update the comment to clarify? 

Would it make sense to return a boolean indicating whether the operation succeeded or not?

Reply:
@mbasmanova 
It says it in the comment: "Set worker into the ACTIVE state if we aren't shutting down.".
It does not make sense to return bool, because we aren't using it.

Reply:
I see. Then it seems to make sense to rename detachWorker to maybeDetachWorker for consistency.